service/routes.py

```diff
@@ -38,16 +38,90 @@ def healthcheck():
+# Configure Swagger before initializing it
+######################################################################
+api = Api(app,
+          version='1.0.0',
+          title='Orders REST API Service',
+          description='This is the back end for an eCommerce web site as a RESTful microservice for the resource order.',
+          default='orders',
+          default_label='Orders operations',
+          doc='/apidocs',
+          prefix='/api'
+          )
+
+# Define the model so that the docs reflect what can be sent
+create_item_model = api.model('Item', {
+    'item_id': fields.Integer(required=True,
+                                 description='Item id of the item'),
+    'item_name': fields.String(required=True,
+                            description='Name of the item'),
+    'item_qty': fields.Integer(required=True,
+                               description='Quantity of the item'),
+    'item_price': fields.Float(required=True,
+                          description='Price of the item')
+})
+
+item_model = api.inherit(
+    'ItemModel', 
+    create_item_model,
+    {
+        'id': fields.String(readOnly=True,
+                            description='The unique id assigned internally by service'),
+    }
+)
+create_model = api.model('Pet', {
+    'customer_id': fields.Integer(required=True,
+                                  description='The customer id of the Order'),
+    'order_items': fields.List(fields.Nested(create_item_model, required=True), required=True,
+                               description='The items in the Order')
+})
+
+order_model = api.inherit(
+    'OrderModel', 
+    create_model,
+    {
+        'id': fields.String(readOnly=True,
+                            description='The unique id assigned internally by service'),
+    }
+)
+
+######################################################################
+#  PATH: /orders/{id}
+######################################################################
+@api.route("/orders/<int:order_id>", strict_slashes=False)
+@api.param('order_id', 'The Order identifier')
+class OrderResource(Resource):
+    """
+    OrderResource class
+    Allows the manipulation of a single Order
+    GET /order{id} - Returns an Order with the id
+    PUT /order{id} - Update an Order with the id
+    DELETE /order{id} -  Deletes an Order with the id
+    """
+
+    # ------------------------------------------------------------------
+    # RETRIEVE AN ORDER
+    # ------------------------------------------------------------------
+    @api.doc('get_orders')
+    @api.response(404, 'Order not found')
+    @api.marshal_with(order_model)
+    def get(self, order_id):
+        """
+        Retrieve a single order
+        This endpoint will return an Order based on it's id
+        """
+        app.logger.info("Request for order with id: %s", order_id)
+        order = Order.find(order_id)
+        if not order:
+            api.abort(status.HTTP_404_NOT_FOUND, "Order was not found.")
```

Comment:
As advised by Professor in class, its better to use the following abort function here, this logs and also aborts 

def abort(error_code: int, message: str):
    """Logs errors before aborting"""
    app.logger.error(message)
    api.abort(error_code, message)